### PR TITLE
test: add metadata process tests and increase coverage

### PR DESCRIPTION
**Pull request by kruskall**
Add testcases for uncompressed, gzip and deflate. Ensure an
error is returned if the data is malformed and increase
code coverage.
Bring coverage to 100% for metadata.go

### CODE REVIEW

File: apm-lambda-extension/apmproxy/metadata_test.go [hunk 1]
```diff
@@ -37,17 +40,106 @@ func Test_processMetadata(t *testing.T) {
...
+		"malformed deflate": {
+			data: func() []byte {
+				var b bytes.Buffer
+
+				w := zlib.NewWriter(&b)
+
+				_, err := w.Write(benchBody)
+				require.NoError(t, err)
+
+				return b.Bytes()
```
Comment (axw) [inline apm-lambda-extension/apmproxy/metadata_test.go:121]:
```suggestion
				// Return the buffer contents without closing the zip.Writer,
				// which will lead to io.ErrUnexpectedEOF.
				return b.Bytes()
```

File: apm-lambda-extension/apmproxy/metadata_test.go [hunk 2]
```diff
@@ -37,17 +40,106 @@ func Test_processMetadata(t *testing.T) {
...
+			encodingType: "deflate",
+		},
+		"invalid deflate": {
+			data: func() []byte {
+				return benchBody
+			},
+			expectError:  zlib.ErrHeader,
+			encodingType: "deflate",
+		},
+		"malformed deflate": {
```
Comment (axw) [inline apm-lambda-extension/apmproxy/metadata_test.go:112]:
```suggestion
		"incomplete deflate": {
```

Might be a little clearer?

**Review by axw — APPROVED** [review]
LGTM!

File: apm-lambda-extension/apmproxy/metadata_test.go (hunk 2 above)
Comment (kruskall) [reply apm-lambda-extension/apmproxy/metadata_test.go]: Agree 👍

**Pull request merged**